python/understack-flavor-matcher/tests/test_flavor_spec.py

```diff
@@ -147,41 +172,73 @@ def flavors():
 
 def test_exact_match(flavors):
     machine = Machine(
-        memory_mb=102400, cpu="AMD EPYC 9254 245-Core Processor", disk_gb=500
+        memory_mb=102400,
+        cpu="AMD EPYC 9254 245-Core Processor",
+        disk_gb=500,
+        model="Dell XPS1319",
     )
     assert flavors[0].score_machine(machine) == 100
     assert flavors[1].score_machine(machine) == 0
 
 
+def test_wrong_model_non_match(flavors):
+    machine = Machine(
+        memory_mb=102400,
+        cpu="AMD EPYC 9254 245-Core Processor",
+        disk_gb=500,
+        model="Some other model",
+    )
+    for flavor in flavors:
+        assert flavor.score_machine(machine) == 0
+
+
 def test_memory_too_small(flavors):
     machine = Machine(
-        memory_mb=51200, cpu="AMD EPYC 9254 245-Core Processor", disk_gb=500
+        memory_mb=51200,
+        cpu="AMD EPYC 9254 245-Core Processor",
+        disk_gb=500,
+        model="Dell XPS1319",
     )
-    assert all(flavor.score_machine(machine) for flavor in flavors) == 0
+    for flavor in flavors:
+        assert flavor.score_machine(machine) == 0
 
 
 def test_disk_too_small(flavors):
     machine = Machine(
-        memory_mb=204800, cpu="AMD EPYC 9254 245-Core Processor", disk_gb=100
+        memory_mb=204800,
+        cpu="AMD EPYC 9254 245-Core Processor",
+        disk_gb=100,
+        model="Dell XPS1319",
     )
     assert all(flavor.score_machine(machine) for flavor in flavors) == 0
 
 
 def test_cpu_model_not_matching(flavors):
-    machine = Machine(memory_mb=102400, cpu="Non-Existent CPU Model", disk_gb=500)
+    machine = Machine(
+        memory_mb=102400,
+        cpu="Non-Existent CPU Model",
+        disk_gb=500,
+        model="Dell XPS1319",
+    )
     assert all(flavor.score_machine(machine) for flavor in flavors) == 0
```

Comment:
Are these asserts doing what it looks like they should be doing?   all() answers a boolean, which `==` compares as a truthy/falsey, so I think this is asserting that at least ONE of the scores is zero, which is all mighty confusing.

Reply:
You are absolutely right - they were indeed matching incorrectly - it was mean to be 

```python
assert all(flavor.score_machine(machine) == 0 for flavor in flavors)
```

addressed in [93a3caa](https://github.com/rackerlabs/understack/pull/481/commits/93a3caa9fb17538405b0285f02b1aa5915801274)